### PR TITLE
maxWidth function is added

### LINKED ISSUE #8586: Filter: Add max-width on filter input box
Re the filter script https://wet-boew.github.io/v4.0-ci/demos/filter/filter-en.html , the filter input box stretches the full width of the element it's inside.  In most cases, that would be the full size of the page 1170px.  This creates poor usability on the box, as inputs should be constrained the roughly the size of what the user would be able to put in the box (it would be like an input that is 1000px wide when all it wants in your phone number).  Can you restrict the size to something like max-width: 400px?

### PR DESCRIPTION
**Pull request by namjohn920**
maxWidth feature is now implemented. 
If user puts maxWidth attributes, the filter's width will change accordingly. 
fixes #8586

### CODE REVIEW

File: src/plugins/filter/filter.js [hunk 1]
```diff
@@ -83,7 +91,8 @@ var componentName = "wb-filter",
 
 			totalEntries = $elm.find( ( settings.section || "" ) + " " + settings.selector ).length;
 
-			filterUI = "<div class=\"input-group\"><label for=\"" + inptId + "\" class=\"input-group-addon\"><span class=\"glyphicon glyphicon-filter\" aria-hidden=\"true\"></span> " + i18nText.filter_label + "</label><input id=\"" + inptId + "\" class=\"form-control " + inputClass + "\" data-" + dtNameFltrArea + "=\"" + elm.id + "\" type=\"search\"></div>" + "<p aria-live=\"polite\" id=\"" + elm.id + "-info\">" + infoFormater( totalEntries, totalEntries ) + "</p>";
+			filterUI = "<div style=\"max-width:" + maxWidth + ";\" class=\"input-group\"><label for=\"" + inptId +
```
Comment (GormFrank) [inline src/plugins/filter/filter.js]:
```suggestion
			filterUI = "<div " + (maxWidth ? "style=\"max-width:" + maxWidth + ";\" : "") + "class=\"input-group\"><label for=\"" + inptId +
```

**Review by GormFrank — CHANGES_REQUESTED** [review]

File: src/plugins/filter/filter.js [hunk 2]
```diff
@@ -37,11 +37,19 @@ var componentName = "wb-filter",
 		var elm = wb.init( event, componentName, selector ),
 			$elm, elmTagName, filterUI, prependUI,
 			settings, setDefault,
-			inptId, totalEntries;
+			inptId, totalEntries,
+			attr,
```
Comment (GormFrank) [inline src/plugins/filter/filter.js]: You do not need to have `attr` if it is only for `maxWidth`. `maxWidth` will do the trick just fine.

File: src/plugins/filter/filter.js [hunk 3]
```diff
@@ -37,11 +37,19 @@ var componentName = "wb-filter",
...
+			maxWidth;
+
 		if ( elm ) {
 			$elm = $( elm );
-
 			elmTagName = elm.nodeName;
+			attr = elm.getAttribute( "maxWidth" );
+
+			if ( attr !== null ) {
```
Comment (GormFrank) [inline src/plugins/filter/filter.js]: Following that logic, you won't need to have this if

**Review by GormFrank — CHANGES_REQUESTED** [review]

**Review by GormFrank — APPROVED** [review]

File: src/plugins/filter/filter.js [hunk 4]
```diff
@@ -37,11 +37,14 @@ var componentName = "wb-filter",
...
 			settings, setDefault,
-			inptId, totalEntries;
+			inptId, totalEntries,
+			maxWidth;
+
 		if ( elm ) {
 			$elm = $( elm );
-
 			elmTagName = elm.nodeName;
+			maxWidth = elm.getAttribute( "maxWidth" );
```
Comment (duboisp) [inline src/plugins/filter/filter.js]: There is no attribute "maxWidth" for the host element that would.

File: src/plugins/filter/filter.js [hunk 5]
```diff
@@ -83,7 +86,8 @@ var componentName = "wb-filter",
 
 			totalEntries = $elm.find( ( settings.section || "" ) + " " + settings.selector ).length;
 
-			filterUI = "<div class=\"input-group\"><label for=\"" + inptId + "\" class=\"input-group-addon\"><span class=\"glyphicon glyphicon-filter\" aria-hidden=\"true\"></span> " + i18nText.filter_label + "</label><input id=\"" + inptId + "\" class=\"form-control " + inputClass + "\" data-" + dtNameFltrArea + "=\"" + elm.id + "\" type=\"search\"></div>" + "<p aria-live=\"polite\" id=\"" + elm.id + "-info\">" + infoFormater( totalEntries, totalEntries ) + "</p>";
+			filterUI = "<div " +  ( maxWidth ? "style=\"max-width:" + maxWidth + ";\"" : "" ) + "class=\"input-group\"><label for=\"" + inptId +
```
Comment (duboisp) [inline src/plugins/filter/filter.js]: We should avoid inline style. Use the plugin SASS

https://github.com/wet-boew/wet-boew/blob/master/src/plugins/filter/base.scss

**Review by duboisp — CHANGES_REQUESTED** [review]

**Comment by duboisp** [comment]
Add 2 breaking point by default:
md device => 80%
lg & xl device => 60%

File: src/plugins/filter/base.scss [hunk 6]
```diff
@@ -1,3 +1,28 @@
...
+
+//Screen size large/x-large
+.wb-filter {
+	.input-group {
+		max-width: 60%;
+	}
+}
+
+//Screen size medium
+@media screen and (max-width: 960px) {
```
Comment (GormFrank) [inline src/plugins/filter/base.scss]: Look at the Tabs plugin for how to deal with media queries in a plugin.

Also, your SASS file name should start with an underscore.

**Review by GormFrank — CHANGES_REQUESTED** [review]

**Review by GormFrank — APPROVED** [review]

**Review by duboisp — APPROVED** [review]

**Pull request merged**